src/chrome/client/chromeDebugAdapter/baseCDAState.ts

```diff
@@ -20,7 +24,19 @@ export abstract class BaseCDAState implements IDebugAdapterState {
     }
 
     public async processRequest(requestName: CommandText, args: unknown): Promise<unknown> {
-        return await this._requestProcessor.processRequest(requestName, args);
+        try {
+            return await this._requestProcessor.processRequest(requestName, args);
+        } finally {
+            if (requestName === 'disconnect') {
+                // Disconnect shuts down the session forcefully after executing the available request handler
+                await this.shutdown();
```

Comment:
Will this kill the process before we have a chance to send the response to the 'disconnect' request?

Reply:
At the moment it has 100ms to send the response: https://github.com/microsoft/vscode-chrome-debug-core/pull/454

Reply:
It seems weird for baseCDAstate to call shutdown like this, why not call it from the actual handler?

Reply:
Based on the Debug session end description here: https://microsoft.github.io/debug-adapter-protocol/overview

"The disconnect request is expected to terminate the debuggee (and any child processes) forcefully"

Makes me think that the disconnect could potentially be sent in any state, to kill the debug adapter. This was my way of making sure that all states know how to disconnect

Reply:
Oh ok. disconnect is probably special in that regard then.